scirpy/_tools/_diversity.py

```diff
@@ -65,11 +100,34 @@ def _shannon_entropy(freq):
     diversity = dict()
     for k in sorted(ir_obs[groupby].unique()):
         tmp_counts = clono_counts.loc[clono_counts[groupby] == k, "count"].values
-        tmp_freqs = tmp_counts / np.sum(tmp_counts)
-        diversity[k] = _shannon_entropy(tmp_freqs)
+        
+        if metric in ["normalized_shannon_entropy", "D50", "DXX"]:
+            # calculate frequencies for these two metrics
+            tmp_freqs = tmp_counts / np.sum(tmp_counts)
+            np.testing.assert_almost_equal(np.sum(tmp_freqs), 1)
+
+            if metric == "normalized_shannon_entropy":
+                diversity[k] = _shannon_entropy(tmp_freqs)
+            elif metric == "D50":
+                diversity[k] = _dxx(tmp_freqs)
+            else:
+                diversity[k] = _dxx(tmp_freqs, perc=perc)   
+        else:
+            # make skbio an optional dependency
+            try:
+                import skbio
+            except ImportError:
+                raise ImportError(
+                    "Using scikit-bio’s alpha diversity metrics requires the installation of "
+                    "`scikit-bio`. You can install it with "
+                    "`pip install scikit-bio`."
+                    )
+            else:
+                # skbio.diversity takes count vectors as input and takes care of unknown metrics
+                diversity[k] = skbio.diversity.alpha_diversity(metric, tmp_counts).values[0]
 
     if inplace:
-        key_added = "alpha_diversity_" + target_col if key_added is None else key_added
+        key_added = f"alpha_diversity_{target_col}_{metric}" if key_added is None else key_added
```

Comment:
I just realized this is becoming a monster of a column name. 

I'm wondering if we should just use `f"{metric}_{target_col}"` instead...